### PR TITLE
Speed up pkg/kubelet/volumemanager/reconciler unit tests

### LINKED ISSUE #91834: Slow pkg/kubelet/volumemanager/reconciler unit tests
**What happened**:
The unit tests in pkg/kubelet/volumemanager/reconciler take over 2 minutes to complete.

**What you expected to happen**:
Unit tests should run quickly so that developers can run them frequently without having to wait a long time for them to complete

**How to reproduce it (as minimally and precisely as possible)**:
```
$ go test ./pkg/kubelet/volumemanager/reconciler
ok  	k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler	125.759s
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.19 latest master branch
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
```
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- Kernel (e.g. `uname -a`):
```
Linux cinlogic-xps13 5.4.0-31-generic #35-Ubuntu SMP Thu May 7 20:20:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:
```
$ go version
go version go1.13.4 linux/amd64
```

### PR DESCRIPTION
**Pull request by wzshiming**
**What type of PR is this?**
/kind cleanup
/kind flake

**What this PR does / why we need it**:
120s (timeout) -> 40s

**Which issue(s) this PR fixes**:

Fixes #91834
xref #98486

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by wzshiming** [comment]
``` log
go test -race -c ./pkg/kubelet/volumemanager/reconciler && stress ./reconciler.test
5s: 0 runs so far, 0 failures
10s: 0 runs so far, 0 failures
15s: 0 runs so far, 0 failures
20s: 0 runs so far, 0 failures
25s: 0 runs so far, 0 failures
30s: 0 runs so far, 0 failures
35s: 0 runs so far, 0 failures
40s: 7 runs so far, 0 failures
45s: 8 runs so far, 0 failures
50s: 8 runs so far, 0 failures
55s: 8 runs so far, 0 failures
1m0s: 8 runs so far, 0 failures
1m5s: 8 runs so far, 0 failures
1m10s: 8 runs so far, 0 failures
1m15s: 8 runs so far, 0 failures
1m20s: 16 runs so far, 0 failures
1m25s: 16 runs so far, 0 failures
1m30s: 16 runs so far, 0 failures
1m35s: 16 runs so far, 0 failures
1m40s: 16 runs so far, 0 failures
1m45s: 16 runs so far, 0 failures
1m50s: 16 runs so far, 0 failures
1m55s: 19 runs so far, 0 failures
2m0s: 24 runs so far, 0 failures
2m5s: 24 runs so far, 0 failures
2m10s: 24 runs so far, 0 failures
2m15s: 24 runs so far, 0 failures
2m20s: 24 runs so far, 0 failures
2m25s: 24 runs so far, 0 failures
2m30s: 24 runs so far, 0 failures
2m35s: 27 runs so far, 0 failures
2m40s: 32 runs so far, 0 failures
2m45s: 32 runs so far, 0 failures
2m50s: 32 runs so far, 0 failures
2m55s: 32 runs so far, 0 failures
3m0s: 32 runs so far, 0 failures
3m5s: 32 runs so far, 0 failures
3m10s: 33 runs so far, 0 failures
3m15s: 38 runs so far, 0 failures
3m20s: 40 runs so far, 0 failures
3m25s: 40 runs so far, 0 failures
3m30s: 40 runs so far, 0 failures
3m35s: 40 runs so far, 0 failures
3m40s: 40 runs so far, 0 failures
3m45s: 41 runs so far, 0 failures
3m50s: 43 runs so far, 0 failures
3m55s: 46 runs so far, 0 failures
4m0s: 48 runs so far, 0 failures
4m5s: 48 runs so far, 0 failures
4m10s: 48 runs so far, 0 failures
4m15s: 48 runs so far, 0 failures
4m20s: 48 runs so far, 0 failures
4m25s: 49 runs so far, 0 failures
4m30s: 51 runs so far, 0 failures
4m35s: 54 runs so far, 0 failures
4m40s: 56 runs so far, 0 failures
4m45s: 56 runs so far, 0 failures
4m50s: 56 runs so far, 0 failures
4m55s: 56 runs so far, 0 failures
5m0s: 56 runs so far, 0 failures
5m5s: 57 runs so far, 0 failures
5m10s: 58 runs so far, 0 failures
5m15s: 62 runs so far, 0 failures
5m20s: 64 runs so far, 0 failures
5m25s: 64 runs so far, 0 failures
5m30s: 64 runs so far, 0 failures
5m35s: 64 runs so far, 0 failures
5m40s: 64 runs so far, 0 failures
5m45s: 65 runs so far, 0 failures
5m50s: 66 runs so far, 0 failures
5m55s: 69 runs so far, 0 failures
6m0s: 72 runs so far, 0 failures
6m5s: 72 runs so far, 0 failures
6m10s: 72 runs so far, 0 failures
6m15s: 72 runs so far, 0 failures
6m20s: 72 runs so far, 0 failures
6m25s: 74 runs so far, 0 failures
6m30s: 76 runs so far, 0 failures
6m35s: 78 runs so far, 0 failures
6m40s: 80 runs so far, 0 failures

```

**Comment by adisky** [comment]
there is already one more PR on the same https://github.com/kubernetes/kubernetes/pull/97955

**Comment by wzshiming** [comment]
> there is already one more PR on the same issue #97955

This PR has not had any feedback for a month.

**Comment by adisky** [comment]
> > there is already one more PR on the same issue #97955
> 
> This PR has not had any feedback for a month.
It has been update last week.

**Comment by wzshiming** [comment]
@adisky 
Oh... sorry, didn't notice!

File: pkg/kubelet/volumemanager/reconciler/reconciler_test.go [hunk 1]
```diff
@@ -1226,7 +1237,7 @@ func Test_UncertainDeviceGlobalMounts(t *testing.T) {
 		for _, tc := range tests {
 			testName := fmt.Sprintf("%s [%s]", tc.name, mode)
 			t.Run(testName+"[", func(t *testing.T) {
-
+				t.Parallel()
```
Comment (liggitt) [inline pkg/kubelet/volumemanager/reconciler/reconciler_test.go:1240]: to run these in parallel, you'll need to capture the mode/tc in local vars:

```
modes := []v1.PersistentVolumeMode{v1.PersistentVolumeBlock, v1.PersistentVolumeFilesystem}
for modeIndex := range modes {
  for tcIndex := range tests {
    mode := modes[modeIndex]
    tc := tests[testIndex]
    testName := fmt.Sprintf("%s [%s]", tc.name, mode)
    t.Run(testName+"[", func(t *testing.T) {
      ...
```

File: pkg/kubelet/volumemanager/reconciler/reconciler_test.go [hunk 2]
```diff
@@ -1407,6 +1419,7 @@ func Test_UncertainVolumeMountState(t *testing.T) {
 		for _, tc := range tests {
```
Comment (liggitt) [inline pkg/kubelet/volumemanager/reconciler/reconciler_test.go:1419]: same [comment](https://github.com/kubernetes/kubernetes/pull/98915/files#r572886717) about capturing mode/tc in local vars

**Comment by wzshiming** [comment]
``` log
go test -race -c ./pkg/kubelet/volumemanager/reconciler && stress ./reconciler.test
5s: 0 runs so far, 0 failures
10s: 0 runs so far, 0 failures
15s: 0 runs so far, 0 failures
20s: 0 runs so far, 0 failures
25s: 0 runs so far, 0 failures
30s: 0 runs so far, 0 failures
35s: 1 runs so far, 0 failures
40s: 4 runs so far, 0 failures
45s: 4 runs so far, 0 failures
50s: 4 runs so far, 0 failures
55s: 4 runs so far, 0 failures
1m0s: 4 runs so far, 0 failures
1m5s: 4 runs so far, 0 failures
1m10s: 4 runs so far, 0 failures
1m15s: 8 runs so far, 0 failures
1m20s: 8 runs so far, 0 failures
1m25s: 8 runs so far, 0 failures
1m30s: 8 runs so far, 0 failures
1m35s: 8 runs so far, 0 failures
1m40s: 8 runs so far, 0 failures
1m45s: 8 runs so far, 0 failures
1m50s: 12 runs so far, 0 failures
1m55s: 12 runs so far, 0 failures
2m0s: 12 runs so far, 0 failures
2m5s: 12 runs so far, 0 failures
2m10s: 12 runs so far, 0 failures
2m15s: 12 runs so far, 0 failures
2m20s: 12 runs so far, 0 failures
2m25s: 16 runs so far, 0 failures
2m30s: 16 runs so far, 0 failures
2m35s: 16 runs so far, 0 failures
2m40s: 16 runs so far, 0 failures
2m45s: 16 runs so far, 0 failures
2m50s: 16 runs so far, 0 failures
2m55s: 16 runs so far, 0 failures
3m0s: 20 runs so far, 0 failures
3m5s: 20 runs so far, 0 failures
3m10s: 20 runs so far, 0 failures
3m15s: 20 runs so far, 0 failures
3m20s: 20 runs so far, 0 failures
3m25s: 20 runs so far, 0 failures
3m30s: 20 runs so far, 0 failures
3m35s: 24 runs so far, 0 failures
3m40s: 24 runs so far, 0 failures
3m45s: 24 runs so far, 0 failures
3m50s: 24 runs so far, 0 failures
3m55s: 24 runs so far, 0 failures
4m0s: 24 runs so far, 0 failures
4m5s: 24 runs so far, 0 failures
4m10s: 25 runs so far, 0 failures
4m15s: 28 runs so far, 0 failures
4m20s: 28 runs so far, 0 failures
4m25s: 28 runs so far, 0 failures
4m30s: 28 runs so far, 0 failures
4m35s: 28 runs so far, 0 failures
4m40s: 28 runs so far, 0 failures
4m45s: 29 runs so far, 0 failures
4m50s: 32 runs so far, 0 failures
4m55s: 32 runs so far, 0 failures
5m0s: 32 runs so far, 0 failures
5m5s: 32 runs so far, 0 failures
5m10s: 32 runs so far, 0 failures
5m15s: 32 runs so far, 0 failures
5m20s: 33 runs so far, 0 failures
5m25s: 36 runs so far, 0 failures
5m30s: 36 runs so far, 0 failures
5m35s: 36 runs so far, 0 failures
5m40s: 36 runs so far, 0 failures
5m45s: 36 runs so far, 0 failures
5m50s: 36 runs so far, 0 failures
5m55s: 37 runs so far, 0 failures
6m0s: 40 runs so far, 0 failures
6m5s: 40 runs so far, 0 failures

```

**Comment by ehashman** [comment]
/hold

since we have another PR before this one -- we can figure out which we want to merge but there's good discussion on the other

**Comment by wzshiming** [comment]
fixed from #99174

**Pull request closed without merging**